### ISSUE TITLE
Q:A about the corrent train and the dataset ? 

### ISSUE THREAD

**Issue by lpscr** [shivammehta25/Matcha-TTS#32]
Hi, after several days of testing, could you please let me know if I am on the right track? I am very new to this and need to ensure that I have understood everything correctly. Thank you very much for your time and all your amazing work!

I am trying to understand how many steps are needed to train for a new language that is not English using IPA.

I have some questions about the dataset. I am using WAV files with a sample rate of 22055Hz in mono. The minimum duration is 1.0 second, and the maximum is 15 seconds. Can I use durations outside this range, like a minimum of 0.5 seconds and a maximum of 25 seconds? What should I consider regarding this? Also, should the WAV files be normalized for sound? Sometimes the audio is low, and I want to ensure the correct level. Currently, I normalize the sound to a range of -1.0 to 1.0. Is this correct?

Regarding the length of the dataset, I tested with 14 hours from a single speaker to see how it goes. Can I use more hours, say around 20 hours? How many hours are recommended for fine-tuning or training from scratch?

In terms of single speaker vs. multi-speaker datasets, if I use multi-speakers, with 4 hours for each of the 10 speakers, totaling about 40 hours, does this enhance the training process? Or is it necessary for each speaker to have a substantial number of hours, similar to the 14 hours for a single speaker?

Regarding the training time, is one day sufficient for training, or is it better to train for a week for improved results? I am using an RTX 4090. How many steps are needed to achieve a good model?

When it comes to pretraining vs. training from scratch, is it better to use a pretrained model or train from scratch for a new language?

Lastly, I have trained for about 180k steps. How many more steps are needed to complete the training process?
![image](https://github.com/shivammehta25/Matcha-TTS/assets/147736764/bdb71bbf-1ee0-4969-9a44-3552726f2cb8)

**Comment by shivammehta25** [comment]
Hello! @lpscr,

I am sorry to have missed the questions. 

> I am trying to understand how many steps are needed to train for a new language that is not English using IPA.

I do not have a concrete answer for this, the prosody improves the longer you train but you should be able to see the difference between generated mel spectrograms and original mel spectrograms on the tensorboard. I generally keep an eye on those during training.

> I have some questions about the dataset. I am using WAV files with a sample rate of 22055Hz in mono. The minimum duration is 1.0 second, and the maximum is 15 seconds. Can I use durations outside this range, like a minimum of 0.5 seconds and a maximum of 25 seconds? What should I consider regarding this? Also, should the WAV files be normalized for sound? Sometimes the audio is low, and I want to ensure the correct level. Currently, I normalize the sound to a range of -1.0 to 1.0. Is this correct?

From past experiments, I have noticed that sometimes smaller durations can be problematic for Monotonic Alignment Search (MAS). But the more the better, so I advise against reducing the minimum duration. But, as long as your GPU can handle longer durations, I would say adding longer durations is better than small utterances during training. 

>Regarding the length of the dataset, I tested with 14 hours from a single speaker to see how it goes. Can I use more hours, say around 20 hours? How many hours are recommended for fine-tuning or training from scratch?

For this again, the more the better. If you just talk about fine-tuning you can try fine-tuning even from just 1 hour of dataset. If training from scratch a rough guess would be anything above 6 hours should be a good start (as long as it has variations in data such that the model can learn to pronounce most phones and their combinations). 

> In terms of single speaker vs. multi-speaker datasets, if I use multi-speakers, with 4 hours for each of the 10 speakers, totaling about 40 hours, does this enhance the training process? Or is it necessary for each speaker to have a substantial number of hours, similar to the 14 hours for a single speaker?

I think this should be enough to train a good enough model. 

> Regarding the training time, is one day sufficient for training, or is it better to train for a week for improved results? I am using an RTX 4090. How many steps are needed to achieve a good model?

In one day you would be able to see decent results but there might be noticeable nuances in prosody which should be fixed in 2-3 days of training. One week sounds too good! I have never trained it for that long but if you have a large dataset you can try training it. 

> When it comes to pretraining vs. training from scratch, is it better to use a pretrained model or train from scratch for a new language?

If you are using a similar phoneset you can use pretrained models and use them as base models to fine-tune them otherwise, train it from scratch. 

> Lastly, I have trained for about 180k steps. How many more steps are needed to complete the training process?

If you are done with experiments and plan to finally train the model, I have noticed very good results after 300k+ iterations. 


Hope this answers your questions. Apologies for the delay. Feel free to continue the discussion or close the issue if you feel like. Enjoy 🍵 Matcha-TTS.
